app/javascript/mastodon/features/ui/index.jsx

```diff
@@ -186,7 +186,7 @@ class SwitchingColumnsArea extends PureComponent {
             {redirect}
 
             {singleColumn ? <Redirect from='/deck' to='/home' exact /> : null}
-            {singleColumn && pathName.startsWith('/deck/') ? <Redirect from={pathName} to={pathName.slice(5)} /> : null}
+            {singleColumn && pathName.startsWith('/deck/') ? <Redirect from={pathName} to={{...this.props.location, pathname: pathName.slice(5)}} /> : null}
```

Comment:
Hm, do we really want to use `this.props.location` in here? iirc it is mutable, would that always behave as expected?

Reply:
I am not sure what else we can use here. The props is injected by `react-router`, so it should contain the latest value?

Reply:
Hm, I guess this is fine. The `pathName` already comes from there anyway. Though I do find the whole pattern a little weird.